canmatrix/dbc.py

```diff
@@ -475,18 +478,24 @@ class FollowUps(object):
                 if temp:
                     receiver = list(map(str.strip, temp.group(11).split(',')))
 
-                    tempSig = Signal(temp.group(1),
-                                     startBit=temp.group(2),
-                                     signalSize=temp.group(3),
-                                     is_little_endian=(int(temp.group(4)) == 1),
-                                     is_signed=(temp.group(5) == '-'),
-                                     factor=temp.group(6),
-                                     offset=temp.group(7),
-                                     min=temp.group(8),
-                                     max=temp.group(9),
-                                     unit=temp_raw.group(10).decode(
-                                         dbcImportEncoding),
-                                     receiver=receiver)
+                    extras = {}
+                    if float_factory is not None:
+                        extras['float_factory'] = float_factory
```

Comment:
Without the above change to default `float_factory` to `None`, I think `float_factory` would always be `not None`.